src/starkware_utils/commitment_tree/patricia_tree/patricia_utils.rs

```diff
@@ -0,0 +1,173 @@
+//! Starkware's Merkle-Patricia tree is based on this representation:
+//! Each node can be one of these:
+//! 1. Empty, with value of 0.
+//! 2. Edge node, with value of hash(bottom_node, edge_path) + edge_length.
+//! 3. Binary node, with value of hash(left, right).
+//!
+//! An edge node represents a path in a maximal subtree with a single non-empty node.
+//! for example, the following is encoded
+//! #    0
+//! #  0   0
+//! # 0 h 0 0
+//! as
+//! (2, 1, h)
+//!
+//! If that maximal subtree is trivial, it is encoded as (0, 0, h) where h is the value of
+//! the leaf or the hash corresponding to that subtree.
+#![cfg(test)]
+
+use std::collections::HashMap;
+
+use cairo_vm::Felt252;
+use num_traits::ToPrimitive;
+
+use crate::storage::storage::HashFunctionType;
+
+#[derive(Clone, Debug)]
+pub struct Node {
+    bottom: Felt252,
+    path: Felt252,
+    length: Felt252,
+}
+
+impl Node {
+    fn empty() -> Self {
+        Self { bottom: Felt252::ZERO, path: Felt252::ZERO, length: Felt252::ZERO }
+    }
```

Comment:
Could be better expressed as a `Default` impl

Reply:
I can add a `Default` implementation, but I'll keep this function for symmetry with the Python code.

Reply:
Nevermind, in this place at least I don't particularly care. This code is only used to compute the root hash using a basic method for use in tests. I don't think it's worth changing anything here.